core/base/document.php

```diff
@@ -896,6 +902,38 @@ public function save( $data ) {
 		return true;
 	}
 
+	private function remove_elements_caching_shortcode( $data ) {
+		if ( ! empty( $data['elements'] ) ) {
+			$data['elements'] = $this->remove_caching_shortcode_from_elements( $data['elements'] );
+		}
+
+		return $data;
+	}
+
+	private function remove_caching_shortcode_from_elements( $elements ) {
+		return array_map( function( $element ) {
+			if ( ! empty( $element['elements'] ) ) {
+				$element['elements'] = $this->remove_caching_shortcode_from_elements( $element['elements'] );
+			}
+
+			if ( ! empty( $element['settings'] ) ) {
```

Comment:
@KingYes 
I see a pattern 
Maybe move the not-empty check to the function?
and rename to try_remove_caching_shortcode_from_settings
this will remove a few lines of code and will encapsulate the logic inside the function 
also 
please add text

Reply:
@TzviRabinovitch Just for clarity, I think you mean, `Please add tests`?